### ISSUE TITLE
Multiple mocks for the same path

### ISSUE THREAD

**Issue by danny-cheung** [h2non/gock#54]
```
package main

import (
	"fmt"
	"gopkg.in/h2non/gock.v1"
	"net/http"
)

func main() {
	defer gock.Off()

	gock.New("https://blah.com").Path("/path").
		AddMatcher(func(request *http.Request, request2 *gock.Request) (b bool, e error) {
			fmt.Println("Matcher001")
			return false, nil
		}).
		Reply(200)

	gock.New("https://blah.com").Path("/path").
		AddMatcher(func(request *http.Request, request2 *gock.Request) (b bool, e error) {
			fmt.Println("Matcher002")
			return false, nil
		}).
		Reply(200)

	gock.New("https://blah.com").Path("/path").
		AddMatcher(func(request *http.Request, request2 *gock.Request) (b bool, e error) {
			fmt.Println("FinalMatcher")
			return true, nil
		}).
		Reply(200)

	http.Get("https://blah.com/path")
}
```

Expected Output:
```
Matcher001
Matcher002
FinalMatcher
```

Actual Output:
```
Matcher001
Matcher001
Matcher001
```

**Comment by danny-cheung** [comment]
This is a variation of Issue #20

**Comment by danny-cheung** [comment]
Using the suggested workaround triggers another (but related) bug.

```
package main

import (
	"fmt"
	"gopkg.in/h2non/gock.v1"
	"net/http"
)

func main() {
	defer gock.Off()

	gock.New("https://blah.com").Path("/path").
		SetMatcher(gock.NewMatcher()).
		AddMatcher(func(request *http.Request, request2 *gock.Request) (b bool, e error) {
			fmt.Println("Matcher001")
			return false, nil
		}).
		Reply(200)

	gock.New("https://blah.com").Path("/path").
		SetMatcher(gock.NewMatcher()).
		AddMatcher(func(request *http.Request, request2 *gock.Request) (b bool, e error) {
			fmt.Println("Matcher002")
			return false, nil
		}).
		Reply(200)

	gock.New("https://blah.com").Path("/path").
		SetMatcher(gock.NewMatcher()).
		AddMatcher(func(request *http.Request, request2 *gock.Request) (b bool, e error) {
			fmt.Println("FinalMatcher")
			return true, nil
		}).
		Reply(200)

	http.Get("https://blah.com/path")
}
```

Expected Output:
```
```

**Comment by danny-cheung** [comment]
In `matcher.go`, `NewMatcher()` copies the package-level `Matchers` slice into the new object.

Slices are just structs that point to an underlying array. https://blog.golang.org/go-slices-usage-and-internals ("Slice Internals").

Therefore, all new Matcher objects will point to the same underlying array as `Matchers`.

`NewMatcher()` should create a whole new slice (via `make()`) and then copy the elements across.

**Comment by danny-cheung** [comment]
Fixed by pr #55
